src/casts/convert-closures.rkt

```diff
@@ -2147,20 +2137,251 @@ TODO We can generate better code in this pass for function casts.
                          (If (Unguarded-Box-Ref (Closure-Ref (Var annon-self) 1))
                              (Closure-Ref (Var annon-self) 0)
                              (Blame (Quote "")))))
-                    (Closure-App (Closure-Code (Var tmp-closure2))
-                                 (Var tmp-closure2)
-                                 '()))))
+                       (Closure-App (Closure-Code (Var tmp-closure2))
+                                    (Var tmp-closure2)
+                                    '()))))
                 (Var annon))
                (Quote-Coercion (Static-Id (Uid "static_fn_crcn" 463)))
-            (Quote 0) (Quote 0) (Quote 0))))) 
-         (Begin
+               (Quote 0) (Quote 0) (Quote 0))))) 
```

Comment:
these 3 zeros input to the `apply-coercion` function are no longer needed I believe. I think they should be replaced with a single occurrence of `do-not-suspend-monotonic-heap-casts`.